pkg/login/social/social.go

```diff
@@ -372,11 +372,37 @@ func (s *SocialBase) SupportBundleContent(bf *bytes.Buffer) error {
 	return nil
 }
 
-func (s *SocialBase) extractRoleAndAdmin(rawJSON []byte, groups []string, legacy bool) (org.RoleType, bool) {
+func (s *SocialBase) extractRoleAndAdminOptional(rawJSON []byte, groups []string) (org.RoleType, bool, error) {
 	if s.roleAttributePath == "" {
-		return s.defaultRole(legacy), false
+		if s.roleAttributeStrict {
+			return "", false, errRoleAttributePathNotSet.Errorf("role_attribute_path not set and role_attribute_strict is set")
+		}
+		return "", false, nil
+	}
+
+	if role, gAdmin := s.searchRole(rawJSON, groups); role.IsValid() {
+		return role, gAdmin, nil
+	} else if role != "" {
+		return "", false, errInvalidRole.Errorf("invalid role: %s", role)
 	}
 
+	if s.roleAttributeStrict {
+		return "", false, errRoleAttributeStrictViolation.Errorf("idP did not return a role attribute, but role_attribute_strict is set")
+	}
+
+	return "", false, nil
+}
+
+func (s *SocialBase) extractRoleAndAdmin(rawJSON []byte, groups []string) (org.RoleType, bool, error) {
+	role, gAdmin, err := s.extractRoleAndAdminOptional(rawJSON, groups)
+	if role == "" {
```

Comment:
question: should we still set the role to default even if an error is returned?

Reply:
fixed so it's only handled after the 2 methods